docs/azure/sdk/resource-management.md

```diff
@@ -0,0 +1,325 @@
+Console.WriteLine($"ResourceGroup: {id.ResourceGroupName}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
+
+#### Casting to the base resource identifier
+
+```csharp
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet";
+// Assume we don't know what type of resource id we have we can cast to the base type
+ResourceIdentifier id = resourceId;
+string property;
+if (id.TryGetSubscriptionId(out property))
+    Console.WriteLine($"Subscription: {property}");
+if (id.TryGetResourceGroupName(out property))
+    Console.WriteLine($"ResourceGroup: {property}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
+
+### Manage Existing Resources By Id
+
+Performing operations on resources that already exist is a common use case when using the management SDK. In this scenario, you usually have the identifier of the resource you want to work on as a string. Although the new object hierarchy is great for provisioning and working within the scope of a given parent, it's a tad awkward when it comes to this specific scenario.  
+
+The following example shows how to access an `AvailabilitySet` object and manage it directly with its ID: 
+
+```csharp
+using Azure.Identity;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.Compute;
+using System;
+using System.Threading.Tasks;
+
+// Code omitted for brevity
+
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Compute/availabilitySets/ws2021availSet";
+// We know the availability set is a resource group level identifier since it has a resource group name in its string
+ResourceGroupResourceIdentifier id = resourceId;
+// We then construct a new armClient to work with
+var armClient = new ArmClient(new DefaultAzureCredential());
+// Next we get the specific subscription this resource belongs to
+Subscription subscription = await armClient.GetSubscriptions().GetAsync(id.SubscriptionId);
+// Next we get the specific resource group this resource belongs to
+ResourceGroup resourceGroup = await subscription.GetResourceGroups().GetAsync(id.ResourceGroupName);
+// Finally we get the resource itself
+// Note: for this last stept in this example, Azure.ResourceManager.Compute is needed
+AvailabilitySet availabilitySet = await resourceGroup.GetAvailabilitySets().GetAsync(id.Name);
+```
+However, this approach required a lot of code and 3 API calls to Azure. The same can be done with less code and without any API calls by using extension methods that we have provided on the client itself. These extension methods allow you to pass in a resource identifier and retrieve a scoped client. The object returned is a *[Resource]Operations* mentioned above, since it has not reached out to Azure to retrieve the data yet.
```

Comment:
```suggestion

However, this approach required a lot of code and three API calls to Azure. The same can be done with less code and without any API calls by using extension methods that we have provided on the client itself. These extension methods allow you to pass in a resource identifier and retrieve a scoped client. The object returned is a `[Resource]Operations` mentioned above, since it has not reached out to Azure to retrieve the data yet.
```